r/src/dataset.cpp

```diff
@@ -58,12 +61,28 @@ std::shared_ptr<ds::SourceFactory> dataset___FSSFactory__Make3(
 }
 
 // [[arrow::export]]
-std::shared_ptr<ds::ParquetFileFormat> dataset___ParquetFileFormat__Make() {
-  return std::make_shared<ds::ParquetFileFormat>();
+std::shared_ptr<ds::ParquetFileFormat> dataset___ParquetFileFormat__Make(List_ options) {
+  auto fmt = ds::ParquetFileFormat::Make();
+  if (options.containsElementNamed("reader_options")) {
+    List_ reader_options = options["reader_options"];
+    if (reader_options.containsElementNamed("use_buffered_stream")) {
```

Comment:
For better or worse, this is a slightly different pattern than the e.g. csv read options (see r/src/csv.cpp). `containsElementNamed` seems fine to use here, but I fear that the lack of type checking will expose us to segfault. It also silently ignores any unsupported (including misspelled) options, which isn't the worst behavior but might cause some surprises. 

I wonder if it would be wise to write a validating function in R to ensure that inputs are the right type. Also would help with the documentation.